modules/standard/Socket.chpl

```diff
@@ -0,0 +1,1282 @@
+}
+
+/*
+  Returns the host address.
+  :return: host address
+  :rtype: `string`
+*/
+proc ipAddr.host throws {
+  return _addressStorage.numericHost();
+}
+
+/*
+  Returns the `port` stored in record.
+  :return: Returns numeric port.
+  :rtype: `uint(16)`
+*/
+proc ipAddr.port throws {
+  return _addressStorage.port();
+}
+
+/* compare ipAddr */
+inline operator !=(in lhs: ipAddr,in rhs: ipAddr) throws {
+  return lhs.family != rhs.family || lhs.host != rhs.host || lhs.port != rhs.port;
+}
+
+/* write ipAddr */
+proc ipAddr.writeThis(f) throws {
+  f.write("(","family:",this.family,",host:",this.host,",port:",this.port,")");
+}
+
+pragma "no doc"
+private extern proc qio_get_fd(fl:qio_file_ptr_t, ref fd:c_int):syserr;
+
+/* Type TCPConn returned from :proc:`connect` */
+type tcpConn = file;
+
+/*
+  Returns the file descriptor associated with socket
+  :return: Returns file descriptor.
+  :rtype: `int(32)`
+*/
+proc tcpConn.socketFd throws {
+  var tempfd:c_int;
+  var err:syserr = ENOERR;
+  on this.home {
+    err = qio_get_fd(this._file_internal, tempfd);
+  }
+  if err then try ioerror(err, "in tcpConn.socketFd");
+  return tempfd;
+}
+
+/*
+  Returns the address of remote socket connection
+  :return: Returns remote address.
+  :rtype: `ipAddr`
+*/
+proc tcpConn.addr throws {
+  return getpeername(this.socketFd);
+}
+
+inline operator !=(in lhs: tcpConn,in rhs: tcpConn) {
+  return lhs.socketFd != rhs.socketFd;
+}
+
+proc tcpConn.writeThis(f) throws {
+  f.write("(","addr:",this.addr,",fd:",this.socketFd,")");
+}
+
+pragma "no doc"
+private extern proc sizeof(e): size_t;
+
+extern "struct event_base" record event_base {};
+extern "struct event" record event {};
+
+extern const EV_TIMEOUT:c_short;
+extern const EV_READ:c_short;
+extern const EV_WRITE:c_short;
+extern const EV_SIGNAL:c_short;
+extern const EV_PERSIST:c_short;
+extern const EV_ET:c_short;
+extern const EV_FINALIZE:c_short;
+extern const EV_CLOSED:c_short;
+extern const EVLOOP_NO_EXIT_ON_EMPTY:c_int;
+
+private extern proc event_base_new():c_ptr(event_base);
+private extern proc event_base_dispatch(base: c_ptr(event_base)):c_int;
+private extern proc event_base_loop(base: c_ptr(event_base), flags: c_int):c_int;
+private extern proc event_base_free(base: c_ptr(event_base));
+private extern proc event_base_got_break(base: c_ptr(event_base)):c_int;
+private extern proc event_base_loopbreak(base: c_ptr(event_base)):c_int;
+private extern proc event_new(base: c_ptr(event_base), fd: c_int, events: c_short, callback: c_fn_ptr, callback_arg: c_void_ptr): c_ptr(event);
+private extern proc event_add(ev: c_ptr(event), timeout: c_ptr(timeval)):c_int;
+private extern proc event_del(ev: c_ptr(event)):c_int;
+private extern proc event_free(ev: c_ptr(event)):c_int;
+private extern proc event_remove_timer(ev: c_ptr(event)):c_int;
+private extern proc evutil_make_socket_nonblocking(fd: c_int):c_int;
+private extern proc libevent_global_shutdown();
+
+extern type pthread_t = c_ulong;
```

Comment:
This one seems odd to me. As far as I can tell, the POSIX standard doesn't say anything about what type `pthread_t` should be. It could be a struct. What happens if it is just `extern type pthread_t` ?

Reply:
I will make the change for it